File: learnst-client/src/models/TicketAnswer.ts
import { Ticket } from "./Ticket";
import { User } from "./User";

export interface TicketAnswer
{
    id?: string;
    content: string;
    ticketId: string;
    ticket?: Ticket;
    authorId: string;
    author?: User;
    createdAt?: string;
}

File: Learnst.Dao/Models/TicketAnswer.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Learnst.Dao.Models;

public class TicketAnswer
{
    public Guid Id { get; set; } = Guid.NewGuid();
    [StringLength(500)] public string Content { get; set; } = string.Empty;
    public Guid TicketId { get; set; }
    [JsonIgnore] public Ticket? Ticket { get; set; }
    public Guid AuthorId { get; set; }
    [JsonIgnore] public User? Author{ get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
